src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CaseNotesApiService.kt

```diff
@@ -91,11 +90,22 @@ class CaseNotesApiService(
   }
 
   private fun getLocationId(person: PersonEntity?): String {
-    return if (person?.location.equals("RELEASED", ignoreCase = true)) {
-      "OUT"
-    } else {
-      organisationService.findOrganisationEntityByName(person?.location!!)?.code!!
+    var location = ""
+    try {
+      location = if (person?.location.equals("RELEASED", ignoreCase = true)) {
+        "OUT"
+      } else {
+        return person?.location?.run {
+          organisationService.findOrganisationEntityByName(this)?.code ?: location
+        } ?: location
+      }
+    } catch (ex: Exception) {
+      log.warn("Error getting location id for ${person?.location}. Will write case-notes with location as null. $ex")
```

Comment:
Could you please amend to using the log.warn("message", exception) format, otherwise you lose the stacktrace.

Reply:
Did a separate log to get the stacktrace, but this is better. Updated